crate/memories/src/sqlite_mem.rs

```diff
@@ -92,34 +65,45 @@ impl<Address, Word> SqliteMemory<Address, Word> {
         })
     }
 
-    /// Returns an `SqliteMemory` instance storing data in a table with the given name
-    /// and connecting to the DB using connections from the given pool.
-    ///
-    /// # Arguments
-    ///
-    /// * `pool` - The pool to use for the memory.
-    /// * `table_name` - The name of the table to create.
-    pub async fn connect_with_pool(
-        pool: Pool,
-        table_name: String,
-    ) -> Result<Self, SqliteMemoryError> {
-        let table_name_clone = table_name.clone();
-        pool.conn(move |conn| conn.execute_batch(&create_table_script(&table_name_clone)))
-            .await?;
-        Ok(Self {
+    /// Returns a new memory instance using a pool of connections to an SQLite
+    /// database.
+    pub async fn new_with_pool(pool: Pool, table_name: String) -> Self {
+        Self {
             pool,
             table_name,
             _marker: PhantomData,
-        })
+        }
     }
-}
 
-impl<Address: Send + Sync, Word: Send + Sync> SqliteMemory<Address, Word> {
+    /// Makes sure the correct table exist in the associated database.
```

Comment:
This comment misled me because by reading it without seeing the signature I would say that the goal is returning a boolean equals "table exists ?"

I would rather suggest smth like :

```suggestion
    /// Creates the memory's table in the associated database if it does not exist.
```

Reply:
Felt okay for me but I will change.